### ISSUE TITLE
[Feature request] Notes for download availability

### ISSUE THREAD

**Issue by Wilker-a** [TeamNewPipe/NewPipe#2478]
i think that SoundCloud media should have in the Download button text, a note showing if the song is originally a free to download, a paid one, or if there was no link at all in the site. — e.g show as _Download (No-link)_ if the upload did not mention any info about its availability nor gave a link, or show as _Download (CC)_ if it is licensed under Creative Commons, or _Download (Free, CC)_ if it is Creative Commons and it has a download link for the free download.

also the application should give the option to block or redirect (to the linked store or whatever) downloads of songs that are paid, with the text note shown in the same way as mentioned above.

_an addition to https://github.com/TeamNewPipe/NewPipe/issues/183#issuecomment-515729231_

**Comment by Stypox** [comment]
I don't think this can be implemented without creating problems: looking for arbitrary links in descriptions is difficult and can lead to security problems if the parsed link leads to a malicious page.

**Comment by Wilker-a** [comment]
i was thinking about just seeing the "free download"|"buy"|whatever buttons that usually appear in some uploads, not the ones in the description.

in any case, i think that the issue with security can be less of a problem if there could be a prompt like FairEmail does when opening any links, so the user can be alerted in case they might misclick or need to know where they're going with the link.

![Screenshot of the prompt popup. The "Browse" button shows another prompt asking to choose which broser the user wants to open with, while "yes" just opens with the default webview.](https://user-images.githubusercontent.com/10854855/62011544-201d8a00-b169-11e9-9ed0-343ca5865880.png)

**Comment by B0pol** [comment]
Yeah
Look at [this music](https://soundcloud.com/shakosh2020/yxeenohdx7ar), you can see a « Free Download » button.
To get this, you can easily have it:
in SouncloudStreamExtractor.java
```java
public isFree() throws ParsingException {
    try {
         return track.getString("purchase_title");
    } catch (Exception ignored) {
         return null;
}
```
if there is not that information, it returns null, such as with [this music](https://soundcloud.com/liluzivert/do-what-i-want-produced-by-maaly-raw-don-cannon)
